lib/errorUtils.js

```diff
@@ -25,10 +25,15 @@ function isSystem (err) {
   }
 }
 
+function causedBy (err, cause) {
+  err.cause = cause
+  return err
+}
+
 function handlePromiseError (err, reply) {
   if (!(err instanceof Error)) {
     var error = new TypeError(`Attempted to reject a promise with a non-error value from type '${typeof err}'`)
-    reply.res.connection.emit('error', error)
+    reply.res.connection.emit('error', causedBy(error, err))
```

Comment:
Do we need to create a function just to use it one time?

Reply:
No, but it shows better the convention (attaching errors) for other devs.